app/models/manageiq/providers/vmware/network_manager/refresh_parser.rb

```diff
@@ -84,40 +120,56 @@ def parse_network_subnet(network)
         :name            => subnet_name(network),
         :ems_ref         => uid,
         :gateway         => network.gateway,
-        :dns_nameservers => [network.dns1, network.dns2],
+        :dns_nameservers => [network.dns1, network.dns2].compact,
         :type            => self.class.cloud_subnet_type,
-        :network_ports   => [],
+        :network_ports   => []
       }
+      new_result[:cidr] = to_cidr(network.gateway, network.netmask)
+      new_result[:dhcp_enabled] = network.dhcp_enabled if network.respond_to?(:dhcp_enabled)
 
       # assign myself to the network
       @data_index.fetch_path(:cloud_networks, network.id)[:cloud_subnets] << new_result
 
       return uid, new_result
     end
 
-    def parse_network_port(net_if)
-      uid = port_id(net_if)
-      vm_uid = net_if[:vm].id
+    def parse_network_router(router)
+      parent_id = router[:parent_net].id
+      uid        = "#{router[:network_id]}---#{parent_id}"
+      new_result = {
+        :type          => self.class.network_router_type,
+        :name          => "Router #{router[:parent_net].name} -> #{router.dig(:net_conf, :networkName)}",
+        :ems_ref       => uid,
+        :cloud_network => @data_index.fetch_path(:cloud_networks, parent_id),
+        :cloud_subnets => []
+      }
+
+      # assign myself to the vapp network
+      @data_index.fetch_path(:cloud_subnets, "subnet-#{router[:network_id]}")[:network_router] = new_result
```

Comment:
Think you want to use `#store_path` here

Reply:
Nice one, thanks!